File: config/rules/opengrep/csharp/lang/cwe-330-weak-random/weak-random-security.cs
using System;
using System.Security.Cryptography;

public class WeakRandomExamples
{
    public void Vulnerable()
    {
        // ruleid: csharp.lang.security.cwe-330.weak-random-security
        var rng = new Random();
        var token = rng.Next();

        // ruleid: csharp.lang.security.cwe-330.weak-random-security
        var rng2 = new System.Random(42);

        // ruleid: csharp.lang.security.cwe-330.weak-random-security
        var value = Random.Shared.Next(100000, 999999);

        // ruleid: csharp.lang.security.cwe-330.weak-random-security
        var d = Random.Shared.NextDouble();
    }

    public void Safe()
    {
        // ok: csharp.lang.security.cwe-330.weak-random-security
        var bytes = RandomNumberGenerator.GetBytes(32);

        // ok: csharp.lang.security.cwe-330.weak-random-security
        var number = RandomNumberGenerator.GetInt32(100000, 999999);

        // ok: csharp.lang.security.cwe-330.weak-random-security
        using var rng = RandomNumberGenerator.Create();
        rng.GetBytes(bytes);
    }
}

File: config/rules/opengrep/python/lang/cwe-330-weak-random/weak-random-security.py
import random
import secrets
import os

# ruleid: python.lang.security.cwe-330.weak-random-security
token = random.random()

# ruleid: python.lang.security.cwe-330.weak-random-security
otp = random.randint(100000, 999999)

# ruleid: python.lang.security.cwe-330.weak-random-security
password_char = random.choice("abcdefghijklmnopqrstuvwxyz0123456789")

# ruleid: python.lang.security.cwe-330.weak-random-security
code = random.choices("ABCDEFGHIJKLMNOPQRSTUVWXYZ", k=6)

# ruleid: python.lang.security.cwe-330.weak-random-security
bits = random.getrandbits(128)

# ok: python.lang.security.cwe-330.weak-random-security
token = secrets.token_hex(32)

# ok: python.lang.security.cwe-330.weak-random-security
token = secrets.token_urlsafe(32)

# ok: python.lang.security.cwe-330.weak-random-security
password_char = secrets.choice("abcdefghijklmnopqrstuvwxyz0123456789")

# ok: python.lang.security.cwe-330.weak-random-security
nonce = os.urandom(16)
